include/TPP/Dialect/Tpp/TppOps.td

```diff
@@ -29,47 +29,63 @@ def TppBRGEMMVNNIMemrefInput : StaticMemRefRankOf<[AnyFloat], [4]>;
 
 // Tpp operands is a scalar float or a static memref with rank 1 or 2.
 def TppOperand : AnyTypeOf<[TppMemRef, AnyFloat]>;
+
 // Tpp operands for VNNI layout.
 def TppVNNIOperand : AnyTypeOf<[TppVNNIMemrefInput, AnyFloat]>;
 
-//===----------------------------------------------------------------------===//
-// AddOp
-//===----------------------------------------------------------------------===//
 
-def Tpp_AddOp : Tpp_Op<"add", [SameTypeOperands]> {
-    let summary = "Element-wise addition.";
-    let description = [{
-        The `tpp.add` operation performs element-wise addition
-        on two-dimensional memref.
+// Class for tpp binary operations.
+class Tpp_BinaryOp<string mnemonic, list<Trait> traits = []> :
+  Tpp_Op<mnemonic, !listconcat(traits, [SameTypeOperands])> {
 
-        Example:
+  let arguments = (ins TppOperand:$lhs, TppOperand:$rhs, TppOperand:$out);
 
-        ```mlir
+  let assemblyFormat = [{
+        `ins` `(` $lhs `:` type($lhs) `,` $rhs `:` type($rhs) `)`
+        `out` `(` $out `:` type($out) `)` attr-dict
+  }];
+}
 
-        tpp.add ins(%1 : memref<2x2xf32>) out(%2: memref<2x2xf32>)
+// Class for tpp.unary operations.
+class Tpp_UnaryOp<string mnemonic, list<Trait> traits = []> :
+  Tpp_Op<mnemonic, !listconcat(traits, [])> {
+  
+  let arguments = (ins TppOperand:$input, TppOperand:$output);
+  
+  let assemblyFormat = [{
+    `ins` `(` $input `:` type($input) `)` `out` `(` $output `:` type($output) `)` 
+    attr-dict
+  }];
+}
 
-        ```
-    }];
+//===----------------------------------------------------------------------===//
+// AddOp
+//===----------------------------------------------------------------------===//
 
-    let arguments = (ins TppOperand:$lhs, TppOperand:$rhs);
+def Tpp_AddOp : Tpp_BinaryOp<"add"> {
+  let summary = "Element-wise addition.";
+  let description = [{
+    The `tpp.add` operation performs element-wise addition
+    on two-dimensional memref C = A + B.
 
-    let assemblyFormat = [{
-        `ins` `(` $lhs `:` type($lhs) `)`
-        `out` `(` $rhs `:` type($rhs) `)` attr-dict
-    }];
+    Example:
 
-    let extraClassDeclaration = [{
-      Value getOutput() { return getRhs(); };
-    }];
+    ```mlir
+
+    tpp.add ins(%1 : memref<2x2xf32>, %2: memref<2x2xf32>) 
+            out(%2: memref<2x2xf32>)
+
+    ```
+  }];
 
-    let hasVerifier = 1;
+  let hasVerifier = 1;
 }
 
 //===----------------------------------------------------------------------===//
 // IdentityOp
 //===----------------------------------------------------------------------===//
 
-def Tpp_IdentityOp : Tpp_Op<"identity"> {
+def Tpp_IdentityOp : Tpp_UnaryOp<"identity"> {
   let summary = "Copies input to output.";
```

Comment:
I think `tpp.identity` also supports broadcasting. If so, can you please update the description to mention about it?

Reply:
I'm not seeing this change in the current state.

Reply:
Added doc here: https://github.com/plaidml/tpp-mlir/pull/258/commits/0e9e2107a4c6c6d0d4014bdd304c870668388ea9

Reply:
Oh, I must have been looking at the wrong time. Github isn't always smart on email links.